app/src/main/java/com/theuhooi/uhooipicbook/ui/monsterlist/MonsterListFragment.kt

```diff
@@ -39,13 +43,18 @@ class MonsterListFragment : Fragment(R.layout.monster_list_fragment) {
         val list = binding.monsterListRecyclerview
         val adapter = MonsterListAdapter()
         list.adapter = adapter
-        viewModel.monsters.observe(viewLifecycleOwner) { monsters ->
-            TransitionManager.beginDelayedTransition(list, Stagger())
-            adapter.submitList(monsters)
-        }
         list.layoutManager = LinearLayoutManager(context)
         binding.viewModel = viewModel
         binding.lifecycleOwner = viewLifecycleOwner
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
```

Comment:
```suggestion
        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
```

```kotlin
viewLifecycleOwner.lifecycleScope.launch
```

lifecycleScope.launchだけで良さそう
それに伴い

```kotlin
viewLifecycleOwner.repeatOnLifecycle
```

viewLifecycleOwnerはいらなそう

Reply:
viewLifecycleOwnerは公式ドキュメントのサンプルコードがそうなっていたから必要なのかと！
ref: https://developer.android.com/topic/libraries/architecture/coroutines?hl=ja

いらないなら消しますー！

Reply:
> 普段は使わないですね

Reply:
> ちょっと確認してみますね

Reply:
> lifecycleScopeの話
> 
> ```kotlin
> val LifecycleOwner.lifecycleScope
>     get() = lifecycle.coroutineScope
> ```
> 
> lifecycleはどこから持ってきてるかというと
> FragmentのinitLifecycleの中で
> 
> ```kotlin
> mLifecycleRegistry = new LifecycleRegistry(this)
> ```
> 
> をしている

Reply:
> 一方viewLifecycleOwnerというと
> FragmentのperformCreateViewで
> 
> ```kotlin
> mViewLifecycleOwner = new FragmentViewLifecycleOwner(this, getViewModelStoreOwner())
> ```
> 
> をしている

Reply:
> じゃあLifecycleRegistryとFragmentViewLifecycleOwnerの中身を見てみる

Reply:
> まずLifecycleRegistryから
> constructorは
> 
> ```kotlin
> LifecycleRegistry(LifecycleOwner provider)
> ```
> 
> であるのでFragment側でのinitializeではLifecycleOwnerを渡していることがわかる
> で、Fragment自体がLifecycleOwnerを継承してるのでthisを渡している
> 
> LifeycleRegistryはここまで

Reply:
> 次にFragmentViewLifecycleOwner
> 
> FragmentViewLifecycleOwner自体もLifecycleRegistryのプロパティは持っているようで、FragmentViewLifecycleOwner#initializeで
> 
> ```kotlin
> mLifecycleRegistry = new LifecycleRegistry(this)
> ```
> 
> をしている
> 
> LifecycleRegistryの引数は上で説明しているので割愛
> 
> FragmentViewLifecycleOwnerはSavedStateRegistryOwnerを継承していて、SavedStateRegistryOwnerがLifecycleOwnerを継承しているのでthisでいいみたい
> 
> でSavedStateRegistryOwner#getSavedStateRegistryをFragmentViewLifecycleOwnerで実装しているようだが、その中を見てみる
> 
> ```kotlin
> public SavedStateRegistry getSavedStateRegistry() {
>         initialize();
>         return mSavedStateRegistryController.getSavedStateRegistry();
> }
> ```
> 
> 一旦initializeは無視してOKで、
> 
> ```kotlin
> mSavedStateRegistryController.getSavedStateRegistry();
> ```
> 
>また新しい登場人物が現れた

Reply:
> あ、待ってこれ死ぬ

Reply:
> というわけで結論としては
> 
> ```kotlin
> viewLifecycleOwner.lifecyceScope
> ```
> 
> と
> 
> ```kotlin
> lifecycleScope
> ```
> 
> は厳密にいうと違うわけだが、今回はただCoroutineを起動したいという目的なわけでSavedStateは関係ないため、後者でいいことはわかった

Reply:
> 以上日高の研究結果です

Reply:
```kotlin
 class LocationFragment: Fragment() {
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        // ...
        viewLifecycleOwner.lifecycleScope.launch {
            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
                locationProvider.locationFlow().collect {
                    // New location! Update the map
                }
            }
        }
    }
}
```

> __Important:__ Fragments should _always_ use the `viewLifecycleOwner` to trigger UI updates. However, that’s not the case for `DialogFragment`s which might not have a `View` sometimes. For `DialogFragment`s, you can use the `lifecycleOwner`.

引用: https://medium.com/androiddevelopers/a-safer-way-to-collect-flows-from-android-uis-23080b1f8bda

Reply:
なんか Fragment にはちゃんと `viewLifecycleOwner` を付けろって書いてあるんですよね

Reply:
> うーん、どこからどう見てもCoroutineScopeを起動するだけなら変わらないように見えるんだよなぁ。。。

Reply:
> しかも中に入っていけば行くほど循環参照してるようにしか見えない。。。
なんだこれ。。。

Reply:
とりあえずあってもなくても問題なく動作しました笑
短いほうがいいから、今回はこれでマージしちゃおうかなぁ